src/components/widgets/SelectWidget.js

```diff
@@ -7,7 +7,9 @@ import {asNumber} from "../../utils";
  * always retrieved as strings.
  */
 function processValue({type, items}, value) {
-  if (type === "array" && items && ["number", "integer"].includes(items.type)) {
+  if (value === "") {
+    return undefined;
```

Comment:
We need to add a warning somewhere in the docs, because if one adds `""` as an enum choice in their schema, it will be automatically converted to `undefined` and drop the property from any parent object - which is now consistent with our text inputs but may be surprising to users.